src/cmd/sanity_test/restore/sharepoint.go

```diff
@@ -13,16 +19,150 @@ func CheckSharePointRestoration(
+		"source_container_name", sourceTree.Name)
+
+	common.CompareDiffTrees[models.Siteable, models.Listable](
 		ctx,
-		ac,
-		drive,
-		envs,
-		// skip permissions tests
+		sourceTree,
+		restoredTree,
 		nil)
+
+	common.Infof(ctx, "Success")
+}
+
+func BuildListsSanitree(
+	ctx context.Context,
+	ac api.Client,
+	siteID string,
+	restoreContainerPrefix, exportFolderName string,
+) *common.Sanitree[models.Siteable, models.Listable] {
+	common.Infof(ctx, "building sanitree for lists of site: %s", siteID)
+
+	site, err := ac.Sites().GetByID(ctx, siteID, api.CallConfig{})
+	if err != nil {
+		common.Fatal(
+			ctx,
+			fmt.Sprintf("finding site by id %q", siteID),
+			err)
+	}
+
+	cfg := api.CallConfig{
+		Select: []string{"id", "displayName", "list", "lastModifiedDateTime"},
+	}
+
+	lists, err := ac.Lists().GetLists(ctx, siteID, cfg)
+	if err != nil {
+		common.Fatal(
+			ctx,
+			fmt.Sprintf("finding lists of site with id %q", siteID),
+			err)
+	}
+
+	lists = getAllowedLists(lists)
+	filteredLists := make([]models.Listable, 0)
+
+	filteredLists = filterListsByPrefix(lists, filteredLists, restoreContainerPrefix)
+
+	rootTreeName := ptr.Val(site.GetDisplayName())
+	// lists get stored into the local dir at destination/Lists/
+	if len(exportFolderName) > 0 {
+		rootTreeName = exportFolderName
+	}
+
+	root := &common.Sanitree[models.Siteable, models.Listable]{
+		Self:        site,
+		ID:          ptr.Val(site.GetId()),
+		Name:        rootTreeName,
+		CountLeaves: len(filteredLists),
+		Leaves:      map[string]*common.Sanileaf[models.Siteable, models.Listable]{},
+	}
+
+	for _, list := range filteredLists {
+		listID := ptr.Val(list.GetId())
+
+		listItems, err := ac.Lists().GetListItems(ctx, siteID, listID, api.CallConfig{})
+		if err != nil {
+			common.Fatal(
+				ctx,
+				fmt.Sprintf("finding listItems of list with id %q", listID),
+				err)
+		}
+
+		m := &common.Sanileaf[models.Siteable, models.Listable]{
+			Parent: root,
+			Self:   list,
+			ID:     listID,
+			Name:   ptr.Val(list.GetDisplayName()),
+			// using list item count as size for lists
+			Size: int64(len(listItems)),
+		}
+
+		root.Leaves[m.ID] = m
+	}
+
+	return root
+}
+
+func getAllowedLists(lists []models.Listable) []models.Listable {
+	filteredLists := make([]models.Listable, 0)
+
+	for _, list := range lists {
+		if !api.SkipListTemplates.HasKey(ptr.Val(list.GetList().GetTemplate())) {
+			filteredLists = append(filteredLists, list)
+		}
+	}
+
+	return filteredLists
+}
+
+func filterListsByPrefix(lists, filteredLists []models.Listable, prefix string) []models.Listable {
+	if strings.Contains(prefix, ",") {
+		prefixes := strings.Split(prefix, ",")
+		for _, p := range prefixes {
+			filteredLists = filterListsByPrefix(lists, filteredLists, p)
+		}
+	}
+
+	for _, list := range lists {
+		listDisplayName := ptr.Val(list.GetDisplayName())
+
+		if strings.HasPrefix(listDisplayName, prefix) {
+			filteredLists = append(filteredLists, list)
+		}
+	}
+
+	return filteredLists
```

Comment:
nit: this func is overcomplicating things by acting recursively when it doesn't need recursion.
```suggestion
func filterListsByPrefix(lists, prefix string) []models.Listable {
        result := []models.Listable{}

	for _, list := range lists {
	        for _, pfx := range strings.Split(prefix, ",") {
		    if strings.HasPrefix(ptr.Val(list.GetDisplayName()), pfx) {
			    result = append(result, list)
			    break
		    }
		}
	}

	return result
```